services/roomService/src/lib/CoveyTownController.ts

```diff
@@ -216,6 +216,10 @@ export default class CoveyTownController {
     if (!finishedPlayer) {
       return false;
     }
+    if (!gaveUp) {
+      finishedPlayer.hasCompletedMaze = true;
+    }
```

Comment:
This is so that if the player has ever completed the maze, it will always stay true. If they give up in a later game they still have completed the maze